src/Discovery/test/Consul.Test/Discovery/ConsulDiscoveryClientTest.cs

```diff
@@ -57,82 +51,82 @@ public async Task AddInstancesToListAsync_AddsExpected()
                         }
                     }
                 }
-            }
+            ]
         };
 
-        Task<QueryResult<ServiceEntry[]>> result = Task.FromResult(queryResult);
-        var clientMoq = new Mock<IConsulClient>();
         var healthMoq = new Mock<IHealthEndpoint>();
-        clientMoq.Setup(c => c.Health).Returns(healthMoq.Object);
-        healthMoq.Setup(h => h.Service("ServiceId", options.DefaultQueryTag, options.QueryPassing, QueryOptions.Default, default)).Returns(result);
-
-        var dc = new ConsulDiscoveryClient(clientMoq.Object, options);
-        var list = new List<IServiceInstance>();
-        await dc.AddInstancesToListAsync(list, "ServiceId", QueryOptions.Default, CancellationToken.None);
-        Assert.Equal(2, list.Count);
-
-        IServiceInstance inst = list[0];
-        Assert.Equal("foo.bar.com", inst.Host);
-        Assert.Equal("ServiceId", inst.ServiceId);
-        Assert.True(inst.IsSecure);
-        Assert.Equal(1234, inst.Port);
-        Assert.Equal(2, inst.Metadata.Count);
-        Assert.Contains("foo", inst.Metadata.Keys);
-        Assert.Contains("secure", inst.Metadata.Keys);
-        Assert.Contains("bar", inst.Metadata.Values);
-        Assert.Contains("true", inst.Metadata.Values);
-        Assert.Equal(new Uri("https://foo.bar.com:1234"), inst.Uri);
-
-        inst = list[1];
-        Assert.Equal("foo1.bar1.com", inst.Host);
-        Assert.Equal("ServiceId", inst.ServiceId);
-        Assert.False(inst.IsSecure);
-        Assert.Equal(5678, inst.Port);
-        Assert.Equal(2, inst.Metadata.Count);
-        Assert.Contains("bar", inst.Metadata.Keys);
-        Assert.Contains("secure", inst.Metadata.Keys);
-        Assert.Contains("foo", inst.Metadata.Values);
-        Assert.Contains("false", inst.Metadata.Values);
-        Assert.Equal(new Uri("http://foo1.bar1.com:5678"), inst.Uri);
+
+        healthMoq.Setup(endpoint => endpoint.Service("ServiceId", options.DefaultQueryTag, options.QueryPassing, QueryOptions.Default, default))
+            .Returns(Task.FromResult(queryResult));
+
+        var clientMoq = new Mock<IConsulClient>();
+        clientMoq.Setup(client => client.Health).Returns(healthMoq.Object);
+
+        var optionsMonitor = new TestOptionsMonitor<ConsulDiscoveryOptions>(options);
+        var discoveryClient = new ConsulDiscoveryClient(clientMoq.Object, optionsMonitor, null, NullLogger<ConsulDiscoveryClient>.Instance);
+
+        var serviceInstances = new List<IServiceInstance>();
+        await discoveryClient.AddInstancesToListAsync(serviceInstances, "ServiceId", QueryOptions.Default, optionsMonitor.CurrentValue, CancellationToken.None);
+        Assert.Equal(2, serviceInstances.Count);
```

Comment:
I think at one point you mentioned some mostly-automated way of switching to FluentAssertions, do you have thoughts on doing that in this PR, or something to come back to later?

Reply:
#931 tracks that. I'd like to address it separately.